mobile-app/cypress/integration/functional/wallet/loans/loans.spec.ts

```diff
@@ -31,50 +31,76 @@ function sendTokenToRandomAddress (tokenId: string, isMax = false): void {
   cy.closeOceanInterface()
 }
 
-function validate50PercentButton (loanTokenSymbol: string, paymentTokenSymbol: string, hasSufficientBalance: boolean = false): void {
+function validate50PercentButton (loanTokenSymbol: string, paymentTokenSymbol: string, loanToPaymentTokenRate: number, hasSufficientBalance: boolean = false): void {
+  const penaltyFee = new BigNumber(1.01)
   cy.getByTestID('50%_amount_button').click()
   if (hasSufficientBalance) {
     cy.getByTestID('payback_input_text_error').should('not.exist')
     cy.getByTestID('button_confirm_payback_loan_continue').should('not.have.attr', 'aria-disabled')
     cy.getByTestID('loan_outstanding_balance').invoke('text').then(text => {
       const outstandingBalance = new BigNumber(text.replace(loanTokenSymbol, '').trim())
       if (loanTokenSymbol === paymentTokenSymbol) {
-        cy.getByTestID('payback_input_text').should('have.value', outstandingBalance.div(2).toFixed(8))
+        cy.getByTestID('payback_input_text').invoke('val').then(text => {
+          checkValueWithinRange(text, outstandingBalance.div(2).toFixed(8), 1)
+        })
+      } else {
+        cy.getByTestID('payback_input_text').invoke('val').then(text => {
+          checkValueWithinRange(text, outstandingBalance.multipliedBy(loanToPaymentTokenRate).multipliedBy(penaltyFee).div(2).toFixed(8), 1)
+        })
       }
       cy.getByTestID('loan_payment_percentage').should('have.text', '50.00%')
     })
-  } else {
-    cy.getByTestID('button_confirm_payback_loan_continue').should('have.attr', 'aria-disabled')
-    cy.getByTestID('available_token_balance').invoke('text').then(text => {
-      const availableBalance = new BigNumber(text.replace(paymentTokenSymbol, '').trim())
-
-      if (!availableBalance.isZero()) {
-        cy.getByTestID('payback_input_text_error').should('have.text', `Insufficient ${paymentTokenSymbol} to pay for the entered amount`)
-      }
-
-      if (loanTokenSymbol === paymentTokenSymbol) {
-        cy.getByTestID('payback_input_text').should('have.value', availableBalance.div(2).toFixed(8))
-      }
-    })
+    return
   }
-}
 
-function validateMaxButtonWith0Balance (): void {
-  const availableBalance = new BigNumber(0)
-  cy.getByTestID('MAX_amount_button').click()
-  cy.getByTestID('button_confirm_payback_loan_continue').should('have.attr', 'aria-disabled')
-  cy.getByTestID('payback_input_text').should('have.value', availableBalance.toFixed(8))
-  cy.getByTestID('loan_payment_percentage').should('have.text', '0%')
+  cy.getByTestID('available_token_balance').invoke('text').then(text => {
+    const availableBalance = new BigNumber(text.replace(paymentTokenSymbol, '').trim())
+    if (availableBalance.isZero()) {
```

Comment:
will this flag be different from the `hasSufficientBalance` param that is passed to the function?

Reply:
yes, different. `hasSufficientBalance` means you can pay the whole loan amount. If it's `false`, you can be paying partially or you don't even have any available balance. `isZero` should disable `continue`

Reply:
how do you know if `hasSufficientBalance` is `true`, isn't it when `availableBalance.isZero()` is `false` ?

Reply:
<img width="596" alt="image" src="https://user-images.githubusercontent.com/13292662/163306608-1cd75056-b0f2-4c33-ba05-1671d92e9355.png">

**hasSufficientBalance = false** = available balance >= loan
**hasSufficientBalance = false** = available balance < loan
If available balance = zero, it doesn't have sufficient balance + continue is disabled which is tested 
If available balance > zero, it doesn't have sufficient balance + continue and can still do partial payment

Maybe I can rename it to: `canPayWholeLoanAmount` to improve readability?

Reply:
Updated! Lemmeknow if you can think of other way to improve it!